[Li Sizhuang] iP

lisizhuang-0121 opened the pull request.

lisizhuang-0121 committed:
Add Rename, Greet, Exit features

Renames chatbot, introduces greeting and exit features.

lisizhuang-0121 committed:
Add Echo Function in Chatbot

lisizhuang-0121 committed:
Add task management feature to add and list tasks

lisizhuang-0121 committed:
Add task class

lisizhuang-0121 committed:
Add task marking and unmarking feature

Modifying the command processor to use switch statements.
Enhance usage of classes to manage classes.
Added error handling features.

lisizhuang-0121 committed:
Amend to comply with coding standards

lisizhuang-0121 committed:
Add inherited Todo, Event and Deadline classes

lisizhuang-0121 committed:
Update command processor for Todo, Event and Deadline functions

lisizhuang-0121 committed:
Add support for ToDos, Events, Deadlines

lisizhuang-0121 committed:
Customization of chatbot name to Soccat

lisizhuang-0121 committed:
Improve chatbot feedback for commands

lisizhuang-0121 committed:
Restructuring for code quality

lisizhuang-0121 committed:
Fix display issue when printing line separator.

James-Hong-Jey commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+    public static void toggleDone(String taskNumber, boolean isDone) {
+        try {
+            int taskId = Integer.parseInt(taskNumber);
+            if (taskId > Task.getTaskCount() || taskId <= 0) {
```
Technically* a magic number, even if it is intuitive as bound checking

James-Hong-Jey commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+            String[] tokens = command.split(" ");
+            System.out.println(line);
+            switch (tokens[0]) {
+            case "list": listTasks();
```
For switch cases, put the result on a new line

James-Hong-Jey commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+        return isDone;
+    }
+
+    public void setDone(Boolean done) {
```
Should be an informative boolean name i.e. isDone

James-Hong-Jey commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,17 @@
+    protected String from;
+    protected String to;
+
+    public Event(String name, String from, String to) {
```
more informative variable names i.e. startDate

spinoandraptos commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+        String task = command.replace(type, "").strip();
+        int taskIndex = Task.getTaskCount();
+        switch(type) {
+        case "todo":
```
indentation would be good for the case statements

spinoandraptos commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+            return;
+        }
+        for(int i=0; i<taskCount; i++) {
+            System.out.print(i+1);
```
i+1 is technically a  magic number, will be good to show it is task index

spinoandraptos commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+                System.out.println("Task out of range!");
+                return;
+            }
+            tasks[taskId-1].setDone(isDone);
```
taskId -1 magic number

spinoandraptos commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,91 @@
+            }
+            tasks[taskId-1].setDone(isDone);
+            System.out.println(tasks[taskId-1]);
+        } catch (NumberFormatException e) {
```
good catch! error handling is well considered

spinoandraptos reviewed (commented):
nice!

lisizhuang-0121 committed:
Improve code quality based on suggestions

lisizhuang-0121 committed:
Add SoccatException class for custom error handling

lisizhuang-0121 committed:
Add error handling features for user input

Catch errors entered by users and provide better feedback for user actions.

lisizhuang-0121 committed:
Improve error handling in task marking

lisizhuang-0121 committed:
Merge branch 'master' into branch-Level-5

lisizhuang-0121 committed:
Merge branch 'branch-Level-5'

lisizhuang-0121 committed:
Organize classes as packages

Place Soccat dependencies into a single soccat package

lisizhuang-0121 committed:
Merge branch 'branch-A-Packages'

lisizhuang-0121 committed:
Implement resizable array for task storage

Removal of redundant taskCount function from Task.java
Implement functions to work with ArrayList in Soccat.java

lisizhuang-0121 committed:
Add deletion feature for tasks.

Usage of new function to delete tasks using the delete command.

lisizhuang-0121 committed:
Implement file management for storage of tasks

Add FileAccess class for centralized management of file read and writes
Update gitignore file to exclude output file.

lisizhuang-0121 committed:
Merge branch 'branch-Level-6'

lisizhuang-0121 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Soccat.java

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+import Soccat.*;
```
Try to import your classes explicitly.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+            if (task.isEmpty()) {
+                System.out.println("Oops, the task for todo cannot be blank!");
+                return;
+            }
```
Try not to have arrow head code. Consider abstracting some methods

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+        case "deadline":
+            try {
+                String[] deadlineTokens = task.split(" /by ");
+                tasks.add(new Deadline(deadlineTokens[0], deadlineTokens[1]));
```
Try not to use magic numbers and define them beforehand

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+    public static void deleteTask(String taskNumber) {
+        try {
+            int taskIndex = Integer.parseInt(taskNumber) - 1;
+            System.out.println("Noted. I have removed this task:");
+            System.out.println("\t" + tasks.get(taskIndex));
```
The logical flow seems off. Try to move these lines below after actually removing the task.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+            return;
+        }
+        int taskIndex;
+        for(int i=0; i<taskCount; i++) {
```
Try to add spaces around the operators

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+        }
+    }
+
+    public static void toggleDone(String taskNumber, boolean isDone) {
```
`toggleDone` does not seem to be an appropriate naming as there is no toggling done in your method. Consider using a more appropriate name.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+        throw new SoccatException();
+    }
+
+    public static void processCommand(String line) {
```
Slightly misleading method naming similar to one above.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+            case "list":
+                listTasks();
+                break;
+            case "todo":
+            case "deadline":
+            case "event":
```
Try to add `//Fallthrough` to indicate that it is the intended functionality.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+            case "delete":
+                deleteTask(tokens[1]);
+                break;
+            }
```
Try to include a `default:` case to ensure every option is considered.

kohnh commented on src/main/java/Soccat.java:
```diff
@@ -0,0 +1,178 @@
+            System.out.println("IO Exception during file creation of " + FILENAME);
+        }
+        tasks = taskFile.getTaskData();
+        String line = "____________________________________________________________";
```
Final strings should be in all caps.

kohnh commented on src/main/java/Soccat/FileAccess.java:
```diff
@@ -0,0 +1,93 @@
+                }
+                dataFileWriter.write(taskType + taskStatus + taskDescription + "\n");
+            }
+        }
```
The `try` should always come with a `catch` block

kohnh commented on src/main/java/Soccat/FileAccess.java:
```diff
@@ -0,0 +1,93 @@
+        updateFile();
+    }
+
+    public FileAccess(String fileName) throws IOException, SoccatException {
```
Try to have your constructor before any methods

kohnh commented on src/main/java/Soccat/FileAccess.java:
```diff
@@ -0,0 +1,93 @@
+        File dataFile = new File(fileName);
+        this.fileObject = dataFile;
+        boolean isCreated = dataFile.exists();
+        // Create new file if uncreated
+        if (!isCreated) {
+            isCreated = dataFile.createNewFile();
+        }
+        // Throw exception if creation failed
```
Try to give more descriptive comments

kohnh reviewed (commented):
Some code quality and coding standards issues. Try not to have everything in the main class

lisizhuang-0121 committed:
Restructure to packages for better code quality

Introduct command package for different commands, parser package for parsing commands and storage package for file access and array storage.

lisizhuang-0121 committed:
Improve file creation features for directories that do not exist

Add feature for creating parent directories for file storage.

lisizhuang-0121 committed:
Update gitignore

Exclude meta-inf

lisizhuang-0121 committed:
Improve project structure with OOP

Split code into different packages for better structure.

lisizhuang-0121 committed:
Add find function for searching tasks

Add support for new find command to look for tasks by keyword(s).

lisizhuang-0121 committed:
Merge pull request #1 from lisizhuang-0121/branch-Level-9

Add find function for searching tasks

lisizhuang-0121 committed:
Add javadoc comments for better code readability

Add comments to most of the classes and public methods.

lisizhuang-0121 committed:
Merge pull request #2 from lisizhuang-0121/branch-A-JavaDoc

Add javadoc comments for better code readability

lisizhuang-0121 committed:
Update tasklist.java for better error handling

lisizhuang-0121 committed:
Update readme file for user guide
